### DISCUSSION TITLE
Unable to install quatro-cli getting ERROR: Failed building wheel for quarto-cli?

### DISCUSSION THREAD

**Discussion by johnsnow09** [quarto-dev/quarto-cli#11553] [Q&A]
### Description

I am trying to install Quarto in Python by using the pip command pip install quarto-cli but getting error:

**Error:**
```
ERROR: Failed building wheel for quarto-cli
Failed to build quarto-cli
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (quarto-cli)
```

![image](https://github.com/user-attachments/assets/52b2b5d0-c89f-447a-ac8d-ae80ac51d9fd)

Using System is Windows 11

Python version: Python 3.12.4

Not sure how to fix it.

**Comment by mcanouil** [comment]
Could you elaborate on what you did and what environment precisely?
Providing only the error is unfortunately not enough.

**Reply by johnsnow09** [reply]
Yes Sure, I will try to provide all the details as much as I am aware of it. If you need anything specific then pls let me know.

1. Well I first installed `!pip install quarto` from vscode and it got installed successfully as shown in below :
```
Installing collected packages: nbclient, quarto
Successfully installed nbclient-0.10.0 quarto-0.1.0
```

2. then to preview the `.qmd` file I got a message to **install quatro** so I tried below command from vscode
```
pip install quarto-cli
```

and this gave me the **error** which I have mentioned while creating this Topic in the post.

**Attaching the complete installation log with error:**

I am using Windows 11 platform
Python 3.12.4

I have also tried the pip command to run as administrator but still getting error.

Please let me know if you need any other info from my side. 

Hope this was helpful !!!

**Reply by mcanouil** [reply]
Are you using PowerShell? Running on WSL? What's your Python environment if any?

**Reply by cscheid** [reply]
@johnsnow09 You copy-pasted way more information in that log than you expected. I think you might want to delete it.

**Reply by johnsnow09** [reply]
First I ran from vscode (jupyter Notebook) and that didn't work so Next I tried with cmd as administrator and got the same error.

Ignore previous file. Attaching the correct file here when ran in Jupyter Notebook (vscode)
[quatro_install_log_error2.txt](https://github.com/user-attachments/files/17939460/quatro_install_log_error2.txt)

Attaching Python environment packages list:
[py env pip list.txt](https://github.com/user-attachments/files/17939557/py.env.pip.list.txt)

**Reply by johnsnow09** [reply]
> @johnsnow09 You copy-pasted way more information in that log than you expected. I think you might want to delete it.

Yeah, thanks for letting me know. I have now attached new one with relevant information only. Please use the latest ones.

**Reply by mcanouil** [reply]
@johnsnow09 please edit your previous message and delete the files which clearly contain sensitive data.

**Reply by johnsnow09** [reply]
> @johnsnow09 please edit your previous message and deleted files which clearly contain sensitive data.

Yes I have edited the previous message and removed the link.

**Reply by mcanouil** [reply]
It seems you are using Anaconda.
Are you using Anaconda on purpose?

**Reply by johnsnow09** [reply]
I just have that from few years. No specific purpose.

**Reply by mcanouil** [reply]
I believe the issue lies in your environment, your Anaconda environment.

I would suggest to create a clean Python environment.

I would also personally suggest to not use Conda/Anaconda, use virtual environment.
The following tool is heavily trending these days: https://docs.astral.sh/

**Reply by johnsnow09** [reply]
I am using virtual environment only but I think in managing environment's conda does get involved somewhere. Not sure though.

Ok  sure, I will create a fresh python environment and will try installation there and thanks for letting me know about https://docs.astral.sh/. Will check this as well. thanks again !

**Reply by mcanouil** [reply]
The log you shared clearly shows Anaconda is at play.
It might be only me, but so far I never met someone without headache and using Conda/Anaconda.

If your setup is few years old, it might appreciate a refresh.

**Reply by johnsnow09** [reply]
Yes Anaconda has been a real pain from so long and finally I have uninstalled it and that freed up around 50 gb space 
but I am still getting the error:
> I have installed Python 3.12.4 restarted machine
> Created New Environment with Python 3.12.4
And I tried to install Quarto cli without installing anything else through cmd `pip install quarto-cli`

Getting same error again. Attaching new logs
[quarto_error_after_anac_unistall.txt](https://github.com/user-attachments/files/17945691/quarto_error_after_anac_unistall.txt)

**Reply by johnsnow09** [reply]
I think all the package installation is going to `pip 24.3.1 from C:\Users\vinee\AppData\Roaming\Python\Python312\site-packages\pip (python 3.12)` this issue is happening because of the paths as when I created virtual environment then my pyvenv.cfg file is pointing to Python in `home = C:\Program Files\Python312` and `executable = C:\Program Files\Python312\python.exe` so all the package installation is still heading to associated directory with these  Python paths and not in the virtual environment so I am still getting the same issue as earlier. I probably need to change the Python path to the virtual environment folder.

Do you have idea about pyvenv.cfg paths ? Below is the contents of my `venv.cfg`
```
home = C:\Program Files\Python312
include-system-site-packages = false
version = 3.12.4
executable = C:\Program Files\Python312\python.exe
command = C:\Program Files\Python312\python.exe -m venv V:\1. R & Python work\Python\path\.venv

```

**Reply by johnsnow09** [reply]
After uninstalling Anaconda I was facing an issue of activating virtual environment [https://stackoverflow.com/questions/77711479/how-to-fix-cant-load-file-activate-ps1-because-script-execution-is-disable](activate) which I have resolved now. 

I have cross checked the location of `pip` and `python` running from location and also there were no other packages installed for this environment. The first package that I have tried to install `pip install quarto-cli` but still getting the same error.

Attaching the latest logs.
[quarto_error_after_power_shell_activate_issue.txt](https://github.com/user-attachments/files/17948740/quarto_error_after_power_shell_activate_issue.txt)

**Reply by cderv** [reply]
You shouldn't need to tweak this file. Using venv, and activating, then installing should work fine. 

Here is an example on windows using **uv**: https://github.com/quarto-dev/quarto-cli/discussions/11380#discussioncomment-11238755

**Reply by johnsnow09** [reply]
Thanks @cderv. Yes it was not an issue due to `pyvenv.cfg` file but it was due to virtual environment not activating and everything was going into global. Something changed after deleting Anaconda as I was not able to use any venv after creating them

**Comment by cscheid** [comment]
I'm not on Windows, but I can't repro this on a fresh virtual environment running Python 3.12.6:

```
$ pip install quarto-cli
...
Building wheels for collected packages: quarto-cli
  Building wheel for quarto-cli (pyproject.toml) ... done
  Created wheel for quarto-cli: filename=quarto_cli-1.6.37-py3-none-any.whl size=211240375 sha256=7327a7869bda6fd773a9271413ebc4eec84dd1a982bacf350535f05c37cfaf7c
  Stored in directory: /Users/cscheid/Library/Caches/pip/wheels/dd/dc/f4/6a9e5d423c23a4593a6aab0b5ea5988189eb1dd3bc6faaae2c
...
$ which quarto
/Users/cscheid/Desktop/daily-log/2024/11/27/pip-test/venv/bin/quarto
$ /Users/cscheid/Desktop/daily-log/2024/11/27/pip-test/venv/bin/quarto check
Quarto 1.6.37
[✓] Checking environment information...
      Quarto cache location: /Users/cscheid/Library/Caches/quarto
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.4.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.46.3: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.6.37
      Path: /Users/cscheid/Desktop/daily-log/2024/11/27/pip-test/venv/lib/python3.12/site-packages/quarto_cli/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.04
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/cscheid/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.6
      Path: /Users/cscheid/Desktop/daily-log/2024/11/27/pip-test/venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3, deno, julia-1.9

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.2
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
```

```

**Reply by cderv** [reply]
I believe this is really conda related. This is the second report and can only be reproduce in conda context.

**Comment by cderv** [comment]
@johnsnow09 as you are using anaconda, this is directly related to this other report: 
- https://github.com/quarto-dev/quarto-cli/discussions/11380

It seems building wheel in the context of installing our quarto-cli package from pipy is clearly problematic in Conda environment context. 

When using conda, I would highly recommend using the conda-forge version: https://anaconda.org/conda-forge/quarto
It is a little behind in release - next stable will be out soon once we'll be able to resolve the bundling at https://github.com/conda-forge/quarto-feedstock/pull/65

Otherwise, I would not use conda environment and install using pip. 

Obviously you can also install quarto without pinning a version in an environment by just installing the correct version your system. The python package bundled are there for users who need to pin a specific version of quarto in time inside a virtual environment. 

Anyhow, I don't know what is happening in Conda context that makes our quarto-cli package fail installed, and it will be hard to solve without knowing. There are other ways to install quarto so I would advice to switch for now. 

Hope it helps

**Reply by johnsnow09** [reply]
I have installed Anaconda again and tried with conda forge and was able to install it now. Not sure if I will get any issues or not when I will try to use it but as of now atleast I was able to install it by using anaconda. 

Thanks alot to the whole team for the quick replies and suggestions. Really appreciate that !!!
